api/src/messaging/api_views.py

```diff
@@ -33,6 +43,41 @@ def get_queryset(self):
         return self.request.user.rooms.all()
 
 
+class MembershipCreateDestroy(APIView):
+    permission_classes = [
+        IsAuthenticated,
+        IsMember,
+    ]
+
+    @swagger_auto_schema(tags=["Messaging"])
+    def post(self, request, *args, **kwargs):
+        member = request.data
+        roomid = kwargs["id"]
+        room = Room.objects.get(id=roomid)
+        if room.members.filter(id=member):
+            return Response({"message": "Member already exists"})
+
+        try:
+            user = CustomUser.objects.get(id=member)
+            room.members.add(user)
+            return Response({"message": "Member Created."})
+
+        except ObjectDoesNotExist:
+            return Response(
+                {"detail": "User not found."}, status=status.HTTP_404_NOT_FOUND
```

Comment:
I think maybe we can use a status 400 instead of 404? Because the room was still found, but the user in the request body was not. I think this is more of a bad request